### PR TITLE
Header test updates

### PR DESCRIPTION
**Pull request by andreiblt1304**
Fix after reviews

### CODE REVIEW

**Review by CostinCarabas — APPROVED** [review]

**Review by axenteoctavian — CHANGES_REQUESTED** [review]

File: header-verifier/tests/header_verifier_blackbox_test.rs
[the remark this answers is not in the document]
Comment (andreiblt1304) [reply header-verifier/tests/header_verifier_blackbox_test.rs:1]: I just realized that there is no checking if the address is an actual SC address, I think it's better to have that inside the endpoint

File: header-verifier/src/lib.rs [hunk 1]
```diff
@@ -67,8 +67,14 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
             "Only ESDT Safe contract can call this endpoint"
         );
 
-        self.pending_hashes(hash_of_hashes)
+        let was_hash_removed = self
```
[the remark this answers is not in the document]
Comment (andreiblt1304) [reply header-verifier/src/lib.rs:70]: i mean, if that point is reached isn't it better to have an actual error message rather than the ugly "bad storage array" one, it will help the user more

File: header-verifier/tests/header_verifier_blackbox_test.rs [hunk 2]
```diff
@@ -110,46 +110,57 @@ impl HeaderVerifierTestState {
...
-
     fn get_bls_keys(&mut self, bls_keys_vec: Vec<ManagedBuffer<StaticApi>>) -> BlsKeys {
-        let bls_keys = bls_keys_vec.iter().map(|key| key.clone().into()).collect();
+        let bls_keys = bls_keys_vec.iter().map(|key| key.clone()).collect();
 
         bls_keys
     }
 
-    fn get_mock_operation(
+    fn generate_bridge_operation_struct(
```
[the remark this answers is not in the document]
Comment (andreiblt1304) [reply header-verifier/tests/header_verifier_blackbox_test.rs:159]: Do you mean that we should consider an `ManagedBuffer` like `operation_1` to be a hash ?

File: header-verifier/tests/header_verifier_blackbox_test.rs [hunk 3]
```diff
@@ -110,46 +110,57 @@ impl HeaderVerifierTestState {
             .run();
     }
 
+    fn propose_remove_execute_hash_wrong_caller(
```
[the remark this answers is not in the document]
Comment (andreiblt1304) [reply header-verifier/tests/header_verifier_blackbox_test.rs:113]: The reason behind this is to use in other error scenarios the `propose_remove_executed_hash` function, it will be easier for further tests.

File: header-verifier/tests/header_verifier_blackbox_test.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (axenteoctavian) [reply header-verifier/tests/header_verifier_blackbox_test.rs:159]: Yes, `operation_`1` is a hash.

File: header-verifier/tests/header_verifier_blackbox_test.rs [hunk 4]
```diff
@@ -110,46 +110,57 @@ impl HeaderVerifierTestState {
...
-
     fn get_bls_keys(&mut self, bls_keys_vec: Vec<ManagedBuffer<StaticApi>>) -> BlsKeys {
-        let bls_keys = bls_keys_vec.iter().map(|key| key.clone().into()).collect();
+        let bls_keys = bls_keys_vec.iter().map(|key| key.clone()).collect();
 
         bls_keys
     }
 
-    fn get_mock_operation(
+    fn generate_bridge_operation_struct(
```
Comment (axenteoctavian) [inline header-verifier/tests/header_verifier_blackbox_test.rs:159]: You change this function but like this is not reusable multiple times.
Just make one function that generates the operations and another one for checking maybe and use both of them in tests.

File: header-verifier/src/lib.rs [hunk 5]
```diff
@@ -67,6 +67,11 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
             "Only ESDT Safe contract can call this endpoint"
         );
 
+        require!(
```
Comment (axenteoctavian) [inline header-verifier/src/lib.rs:70]: Please remove this and keep it like it was before. When the time comes and we will look for possible security issues and audits we'll cover edge cases for this function as well, but for now it's good without this require.

File: header-verifier/tests/header_verifier_blackbox_test.rs
[the remark this answers is not in the document]
Comment (axenteoctavian) [reply header-verifier/tests/header_verifier_blackbox_test.rs:1]: I don't see any change after your comment

[the remark this answers is not in the document]
Comment (axenteoctavian) [reply header-verifier/tests/header_verifier_blackbox_test.rs:1]: Also you don't even deploy the `ENSHRINE_CODE_PATH`, you deploy 2 of `HEADER_VERIFIER_ADDRESS`.
Can't you just register an empty contract? There is no need to deploy a good contract, you need just an empty mock and not create other dependencies.

**Review by axenteoctavian — CHANGES_REQUESTED** [review]

File: header-verifier/src/lib.rs (hunk 5 above)
Comment (andreiblt1304) [reply header-verifier/src/lib.rs:70]: As my development cycle, every endpoint that I will write I will also try to cover all the edge cases with `requires`. Afterwards through testing I will add further `requires`. I thought that having a more verbose error message will help in the grand scheme of the execution phase. In this case I will remove it and more `requires` when we realize that they are actually needed.

File: header-verifier/tests/header_verifier_blackbox_test.rs
[the remark this answers is not in the document]
Comment (andreiblt1304) [reply header-verifier/tests/header_verifier_blackbox_test.rs:1]: For the test to be 100% relevant we need:
1. A mocked deployed contract (ESDT-Safe)
2. ESDT-Safe has to call Header-Verifier to execute `remove_executed_hash`
3. Add to the `remove_execute_hash` endpoint a check for the caller to be an actual SC address

There are no "empty" testing SCs inside the framework, the easier route would be to use a normal `TestAddress` that calls that endpoint since the endpoint doesn't check for the caller to be a SC address (IMO it should)

File: header-verifier/tests/header_verifier_blackbox_test.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (axenteoctavian) [reply header-verifier/tests/header_verifier_blackbox_test.rs:102]: Just delete this and use `propose_remove_executed_hash` instead, then should be good

**Review by axenteoctavian — CHANGES_REQUESTED** [review]

**Review by axenteoctavian — APPROVED** [review]
